checker.cpp

```diff
@@ -3,44 +3,84 @@
 #include <unistd.h>
 using namespace std;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if(temperature > 102 || temperature < 95) {
-    cout << "Temperature critical!\n";
+void PrintCriticalMessage(string message) {
+    cout << message << endl;
     for (int i = 0; i < 6; i++)
     {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+        cout << "\r* " << flush;
+        sleep(1);
+        cout << "\r *" << flush;
+        sleep(1);
     }
+}
 
-    return 0;
-  } else if(pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+void PrintWarningMessage(){
+  cout <<"Approaching hypothermia" <<endl;
+}
+bool istemperatureCritical(float temperature){
+    return temperature > 102 || temperature < 95;
+}
+bool isTemperatureWarning(float temperature) {
+    float warningTolerance = 102 * 0.015;
+    return temperature >= 102 - warningTolerance || temperature <= 95 + warningTolerance;
+}
+
+bool checkTemperature(float temperature) {
+
+    if (isTemperatureWarning(temperature)) {
+        PrintWarningMessage();
     }
-    return 0;
-  } else if(spo2 < 90) {
-    cout << "Oxygen Saturation out of range!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+    else if (istemperatureCritical(temperature)) {
+        PrintCriticalMessage("Temperature critical!");
+        return false;
+    }
+    return true;
+}
+
+
+bool ispulseRateCritical(float pulseRate){
+    return pulseRate < 60 || pulseRate > 100;
+}
+bool ispulseRateWarning(float pulseRate)
+{   
+    float warningTolerance = 100 * 0.015;
+    return pulseRate >= 100 - warningTolerance || pulseRate <= 60 + warningTolerance;
+}
+
+bool checkPulseRate(float pulseRate) {
+    
+    if (ispulseRateWarning(pulseRate)) {
+        PrintWarningMessage();
```

Comment:
This will say `Approaching hypothermia`, even though the issue is about the pulse rate?